main.c

```diff
@@ -1295,8 +1296,22 @@ int main(int argc, char **argv) {
 		create_surface(surface);
 	}
 
+	if (state.ext_session_lock_manager_v1) {
+		while (!state.locked) {
+			if (wl_display_dispatch(state.display) < 0) {
+				swaylock_log(LOG_ERROR, "wl_display_dispatch() failed");
+				return 2;
+			}
+		}
+	} else {
+		if (wl_display_roundtrip(state.display) < 0) {
+			swaylock_log(LOG_ERROR, "wl_display_roundtrip() failed");
+			return 2;
+		}
+		state.locked = true;
```

Comment:
I don't think this statement is correct; this should only be set when the locked event has been received from the compositor (as done in `ext_session_lock_v1_handle_locked`).

Reply:
This is the legacy code-path, when ext-session-lock-v1 is unused (ie, when layer-shell is used). I've only included it so that this patch can be included in a bugfix release, we'll soon drop that logic: #266.